### PR TITLE
[Api] update and delete comments

### PR DESCRIPTION
**Pull request by ghost**
## Description
* Api  update and delete comments
* Handle ModelNotFound exception
* Handle AuthenticationException
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-khadinh** [comment]
LGTM

**Comment by at-khadinh** [comment]
LGTM

File: app/Http/Controllers/Api/User/CommentController.php [hunk 1]
```diff
@@ -30,4 +33,46 @@ public function index(Post $post, Request $request)
...
+     */
+    public function update(Comment $comments, UpdateCommentRequest $request)
+    {
+        $user = Auth::user();
+        if ($user->id == $comments->user_id) {
+            $comments->content = $request->content;
+            $comments->load('user.userinfo');
+            $comments->save();
+        } else {
+            throw new AuthenticationException();
```
Comment (at-huenguyen) [inline app/Http/Controllers/Api/User/CommentController.php]: ném exception ra đây, rồi chỗ nào bắt lại em?

**Review by ghost — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Api/User/CommentController.php (hunk 1 above)
Comment (at-phuongvu) [reply app/Http/Controllers/Api/User/CommentController.php]: `Handler` em xử lý rồi ạ ...

Comment (at-huenguyen) [reply app/Http/Controllers/Api/User/CommentController.php]: chị tin em :v

**Review by ghost — APPROVED** [review]

**Pull request merged**